google/cloud/datastore/_http.py

```diff
@@ -181,17 +181,25 @@ def lookup(self, project_id, keys, read_options=None):
         )
 
     def run_query(
-        self, project_id, partition_id, read_options=None, query=None, gql_query=None
+        self,
+        project_id=None,
+        partition_id=None,
+        read_options=None,
+        query=None,
+        gql_query=None,
+        request=None,
```

Comment:
This class is not part of the public API for the library, and its only caller is `google.cloud.datastore.query.Iterator._next_page`, which passes only `request` as a dict, and optionally `retry` / `timeout`.  So, we should simplify:
```python
    def run_query(self, request, retry=None, timeout=None):
```